File: src/main/java/net/kigawa/kutil/kutil/file/FileExtension.java
package net.kigawa.kutil.kutil.file;

/**
 * 拡張子の管理のためのクラス
 * to manage file extension
 */
@SuppressWarnings("unused")
public enum FileExtension
{
    LOG("log"),
    /**
     * @deprecated
     */
    @Deprecated
    log("log");

    private final String extension;

    FileExtension(String extension) {
        this.extension = extension;
    }

    /**
     * 拡張子の取得
     * get extension
     *
     * @return extension
     */
    public String getExtension() {
        return "." + extension;
    }

    /**
     * 拡張子を追加して返す
     * add extension and return
     *
     * @param name file name
     * @return added file name
     */
    public String addExtension(String name) {
        return name + getExtension();
    }

    /**
     * 拡張子を追加する
     * add extension
     *
     * @param stringBuffer file name
     * @return added string buffer
     */
    public StringBuffer addExtension(StringBuffer stringBuffer) {
        return stringBuffer.append(".").append(extension);
    }
}

File: kutil/src/commonMain/kotlin/net/kigawa/kutil/kutil/api/file/FileExtension.kt
package net.kigawa.kutil.kutil.api.file;

/**
 * 拡張子の管理のためのクラス
 * to manage file extension
 */
@Suppress("unused", "MemberVisibilityCanBePrivate")
enum class FileExtension(
  private val extension: String,
) {
  LOG("log"),
  ;


  /**
   * 拡張子の取得
   * get extension
   *
   * @return extension
   */
  fun getExtension(): String {
    return ".$extension";
  }

  /**
   * 拡張子を追加して返す
   * add extension and return
   *
   * @param name file name
   * @return added file name
   */
  fun addExtension(name: String): String {
    return name + getExtension();
  }

  /**
   * 拡張子を追加する
   * add extension
   *
   * @param sb file name
   * @return added string buffer
   */
  fun addExtension(sb: StringBuilder): StringBuilder {
    return sb.append(".").append(extension);
  }
}
